pkg/web/client.go

```diff
@@ -192,10 +191,17 @@ func (c *Client) versionFromManifest(image string, weights []File, files []File)
 	return &version, nil
 }
 
-func newVersionURL(username string, image string) url.URL {
+func newVersionURL(image string) (url.URL, error) {
+	imageComponents := strings.Split(image, "/")
 	newVersionUrl := webBaseURL()
-	newVersionUrl.Path = strings.Join([]string{"", username, image, "versions"}, "/")
-	return newVersionUrl
+	if len(imageComponents) != 3 {
+		return newVersionUrl, errors.New("The image URL must have 3 components in the format of " + global.ReplicateRegistryHost + "/username/modelname")
+	}
+	if imageComponents[0] != global.ReplicateRegistryHost {
+		return newVersionUrl, errors.New("The image name must have the " + global.ReplicateRegistryHost + " prefix in fast push.")
```

Comment:
Do users know what "fast push" is? Should this say something like 
```suggestion
		return newVersionUrl, errors.New("The image name must have the " + global.ReplicateRegistryHost + " prefix when using --x-fast.")
```